store/sqlite/sqlite.go

```diff
@@ -405,12 +405,24 @@ func (d db) DeleteGuestLink(id types.GuestLinkID) error {
 	return tx.Commit()
 }
 
-// Purge clears orphaned rows from the database.
+// Purge deletes expired entries and clears orphaned rows from the database.
 func (d db) Purge() error {
+	log.Printf("deleting expired entries from database")
+
+	_, err := d.ctx.Exec(`
+	DELETE FROM
+		entries
+	WHERE
+		entries.expiration_time IS NOT NULL AND entries.expiration_time < CURRENT_TIMESTAMP;`)
+	if err != nil {
+		return err
+	}
+
 	log.Printf("purging orphaned rows from database")
 
 	// Delete rows from entries_data if they don't reference valid rows in
-	// entries. This can happen if the entry insertion fails partway through.
+	// entries. This can happen if the entry insertion fails partway through
```

Comment:
Deleting orphaned rows is something that happens when we have an unpreventable error. It would ideally never happen, except that [adding file data in a SQLite transaction](https://github.com/mtlynch/picoshare/pull/293) takes up too much RAM.

I'd rather not expand it to be the normal way that we delete rows from the `entries_data` table. Can we instead explicitly delete the expired rows from `entries` and `entries_data` rather than orphaning `entries_data` rows and then letting orphan cleanup handle them?

Could we instead explicitly delete